tests/ui/multiple_bound_locations.rs

```diff
@@ -0,0 +1,49 @@
+#![warn(clippy::multiple_bound_locations)]
+
+fn ty<F: std::fmt::Debug>(a: F)
+//~^ ERROR: bound is defined in more than one place
+where
+    F: Sized,
+{
+}
+
+fn lifetime<'a, 'b: 'a, 'c>(a: &'b str, b: &'a str, c: &'c str)
+//~^ ERROR: bound is defined in more than one place
+where
+    'b: 'c,
+{
+}
+
+fn ty_pred<F: Sized>()
+//~^ ERROR: bound is defined in more than one place
+where
+    for<'a> F: Send + 'a,
+{
+}
+
+struct B;
```

Comment:
What if B had generics, say one type and one lifetime, just to mix it up a bit?

Reply:
Not sure to see what you want to test here. ^^'

Reply:
Say what if you had `struct B<F>(F);` and then an `impl<F> B<F> { fn foo(_f: F) -> Self where F: std::fmt::Display { todo!() } }`?

Reply:
It won't the lint but I can add a check for it if you want?

Reply:
Added it.